doc/reference/trd104-syscalls.md

```diff
@@ -496,14 +496,24 @@ failure variant of `Failure`, with an associated error code of
 handle userspace/kernel mismatches should be able to handle `Failure` in
 addition to the expected failure variant (if different than `Failure`).
 
+
 4.3.1 Command Identifier 0
---------------------------------
+--------------------------
+
+ - [Design RFC](../rfcs/2023-08-18--CommandZeroSemantics.md).
+
+Command Identifier 0 provides an existence check for drivers. If a
+driver is accessible to calling application, the kernel will return
```

Comment:
Why is the text rewritten to say the exact same thing? It doesn't constrain the Success type, which was the start of this whole discussion? It is not using keywords.  This is also prescribing *who* returns the `Success`, which this as an API specification shouldn't do (I agree the current text is not great on this either). I would suggest:

Command Identifier 0 provides an existence check for drivers. A return value of `Success` indicates that the driver is present and the userspace process can issue system calls to it. If the driver is not accessible, Command Identifier 0 returns `Failure` with an error code of `ENODEVICE`. A driver may be not accessible because the kernel does not have it, the process does not have the required permissions to use it, or other reasons.

Reply:
Updated with this text √